### PR TITLE
feat: qna provision

### PR DESCRIPTION
**Pull request by natalgar**
## Description

QnA Provisioning

## Task Item

refs #7629

### CODE REVIEW

File: extensions/azurePublishNew/src/node/azureResources/qna.ts [hunk 1]
```diff
@@ -13,26 +32,234 @@ export const qnaDefinition: ResourceDefinition = {
...
+          },
+        ],
+      },
+      enabled: true,
+    });
+  };
+
+  return async (config: QnAResourceConfig, workingSet: ProvisionWorkingSet): Promise<ProvisionWorkingSet> => {
+    // initialize the name
+    const qnaMakerSearchName = `${config.name}-search`.toLowerCase().replace('_', '');
```
Comment (hatpick) [inline extensions/azurePublishNew/src/node/azureResources/qna.ts:204]: it only replaces one of the underscores, is that what you want?

File: extensions/azurePublishNew/src/node/azureResources/qna.ts [hunk 2]
```diff
@@ -13,26 +32,234 @@ export const qnaDefinition: ResourceDefinition = {
...
+    const qnaMakerWebAppName = `${config.name}-qnahost`.toLowerCase().replace('_', '');
```
Comment (hatpick) [inline extensions/azurePublishNew/src/node/azureResources/qna.ts:205]: same here

File: extensions/azurePublishNew/src/node/azureResources/qna.ts [hunk 3]
```diff
@@ -13,26 +32,234 @@ export const qnaDefinition: ResourceDefinition = {
...
+    const qnaMakerServiceName = `${config.name}-qna`;
+
+    const qnaServicePlanName = config.resourceGroupName;
+    const qnaAppInsightsName = config.resourceGroupName;
+    // only support westus in qna
+    if (config.location !== 'westus') {
```
Comment (hatpick) [inline extensions/azurePublishNew/src/node/azureResources/qna.ts:211]: should this be handled on the caller side with a warning bubbled up to the UX?

File: extensions/azurePublishNew/src/node/azureResources/qna.ts [hunk 4]
```diff
@@ -13,26 +32,234 @@ export const qnaDefinition: ResourceDefinition = {
...
   group: COGNITIVE_SERVICES_GROUP_NAME,
+  dependencies: [AzureResourceTypes.RESOURCE_GROUP, AzureResourceTypes.WEBAPP],
 };
 
-const getQnAProvisionMethod = (): ProvisionMethod => {
-  return (config: QnAConfigNew, workingSet: ProvisionWorkingSet): Promise<ProvisionWorkingSet> => {
-    const provisionResult = {};
+type ServiceCredentials = {
+  signRequest: <T>(webResource) => Promise<T>;
+};
```
[the remark this answers is not in the document]
Comment (natalgar) [reply extensions/azurePublishNew/src/node/azureResources/qna.ts]: updated

File: extensions/azurePublishNew/src/node/azureResources/qna.ts [hunk 5]
```diff
@@ -13,26 +32,234 @@ export const qnaDefinition: ResourceDefinition = {
...
+  ) => {
+    return await cognitiveServicesManagementClient.accounts.create(config.resourceGroupName, qnaMakerServiceName, {
+      kind: 'QnAMaker',
+      sku: {
+        name: config.sku ?? 'S0',
+      },
+      location: config.location,
+      properties: {
+        apiProperties: {
+          qnaRuntimeEndpoint: `https://${qnaEndpoint}`,
```
[the remark this answers is not in the document]
Comment (natalgar) [reply extensions/azurePublishNew/src/node/azureResources/qna.ts:115]: lots of good points on this and the others that I will address in a separate PR.

**Review by GeoffCoxMSFT — APPROVED** [review]

**Comment by natalgar** [comment]
replacing this PR with #8321

**Pull request closed without merging**